cdap-tms/src/main/java/co/cask/cdap/messaging/subscriber/AbstractMessagingSubscriberService.java

```diff
@@ -214,8 +228,11 @@ private long fetchAndProcessMessages() throws TopicNotFoundException, IOExceptio
     startTime = System.currentTimeMillis();
 
     // Process the notifications and record the message id of where the processing is up to.
+    // 90% of the tx timeout is .9 * 1000 * txTimeoutSeconds = 900 * txTimeoutSeconds
+    long timeBoundMillis = 900L * txTimeoutSeconds;
+    TimeBoundIterator<Message> timeBoundMessages = new TimeBoundIterator<>(messages.iterator(), timeBoundMillis);
```

Comment:
Should this go to the transaction? Otherwise if something is wrong in the transaction and we are retrying, the time will not start from 0 in the retried transaction.

Reply:
good point